module_1/src/CustomizedMovieForm/CustomizedMovieForm.js

```diff
@@ -0,0 +1,41 @@
+import React, {useEffect, useState} from 'react';
+import Dialog from '../Dialog/Dialog';
+import MovieForm from '../MovieForm/MovieForm';
+import { useNavigate, useParams } from 'react-router-dom';
+
+
+function CustomizedMovieForm () {
+    const [movieData, setMovieData] = useState();
+    const navigate = useNavigate();
+    const {movieId} = useParams();
+    const title = movieId? 'EDIT MOVIE': 'ADD MOVIE';
+
+    useEffect(() => {
+        if (movieId) {
+            fetch(`http://localhost:4000/movies/${movieId}`)
+            .then((response) => response.json())
+            .then((json)=>{
+                setMovieData(json);
+                console.log('movieData', movieData);
+            });
+        }
+    }, []);
```

Comment:
Please pay attention to the React hook rule. For most of the time it's useless to add all callbacks to useEffect dependencies, but it is better to explicitly add commentaries with `// eslint-disable-next-line react-hooks/exhaustive-deps` to make sure it wasn't left so by mistake and get rid of warnings in console. For this concrete hook you should add movieId to dependencies. It should get new movie data when you select another movie (and it is possible both  from your UI and from manually changing parameter in URL)